src/AzureStorage/Queues/AzureStorageQueue.cs

```diff
@@ -153,17 +169,17 @@ public class AzureStorageQueue<T> : QueueBase<T> where T : class {
             _workerErrorCount = 0;
         }
 
-        public override void StartWorking(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
+        protected override void StartWorkingImpl(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
             if (handler == null)
                 throw new ArgumentNullException(nameof(handler));
-
+            
             var linkedCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_queueDisposedCancellationTokenSource.Token, cancellationToken).Token;
 
             Task.Run(async () => {
```

Comment:
I don't think StartWorking really shouldn't be awaited because its non-deterministic and is really meant to just kick off a long running background worker. It has a cancellationtoken that can be used to stop the worker. We already debated about this and decided on this solution, but I am definitely open to discussing it some more.